src/PAModelTests/EntropyTests.cs

```diff
@@ -154,5 +155,101 @@ public void TestPCFControlWillFallBackToControlTemplate(string appName)
+            // Checking if the HostType Entry is added
+            Assert.IsTrue(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostTypePropertyName));
+            Assert.IsFalse(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostServicePropertyName));
+        }
+
+        [DataTestMethod]
+        [DataRow("HostControlTestWithHostTypeAndHostService.msapp")]
+        public void TestHostControlWithHostTypeAndService(string appName)
+        {
+            var root = Path.Combine(Environment.CurrentDirectory, "Apps", appName);
+            Assert.IsTrue(File.Exists(root));
+
+            (var msapp, var errors) = CanvasDocument.LoadFromMsapp(root);
+            errors.ThrowOnErrors();
+
+            Assert.IsTrue(msapp._entropy.HostTypeEntry.Count > 0);
+            Assert.IsTrue(msapp._entropy.HostServiceEntry.Count > 0);
+
+            Assert.IsTrue(msapp._screens.TryGetValue("App", out var app));
+            var sourceFile = IRStateHelpers.CombineIRAndState(app, errors, msapp._editorStateStore, msapp._templateStore, new UniqueIdRestorer(msapp._entropy), msapp._entropy);
+
+            Assert.AreEqual("Host", sourceFile.Value.TopParent.Children.Last().Name);
+            // Checking if the HostType and HostService Entries were added
+            Assert.IsTrue(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostTypePropertyName));
+            Assert.IsTrue(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostServicePropertyName));
+        }
+
+        [DataTestMethod]
+        [DataRow("HostControlTestWithHostType.msapp")]
+        public void TestHostControlWithHostTypeFallBackToTemplate(string appName)
+        {
+            var root = Path.Combine(Environment.CurrentDirectory, "Apps", appName);
+            Assert.IsTrue(File.Exists(root));
+
+            (var msapp, var errors) = CanvasDocument.LoadFromMsapp(root);
+            errors.ThrowOnErrors();
+
+            Assert.IsTrue(msapp._entropy.HostTypeEntry.Count > 0);
+
+            // Clear out the HostType entropy entries
+            msapp._entropy.HostTypeEntry.Clear();
+            Assert.IsTrue(msapp._entropy.HostTypeEntry.Count == 0);
+
+            // Repack the app and validate it matches the initial msapp
+            using (var tempFile = new TempFile())
+            {
+                MsAppSerializer.SaveAsMsApp(msapp, tempFile.FullPath, new ErrorContainer());
+                Assert.IsTrue(MsAppTest.Compare(root, tempFile.FullPath, Console.Out));
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow("HostControlTestWithHostTypeAndHostService.msapp")]
+        public void TestHostControlWithHostTypeAndServiceFallBackToTemplate(string appName)
```

Comment:
nit: Is this test name (and the one above it) accurate? It doesn't fall back to the template if it is missing from Entropy, does it?

Reply:
You are right it does not fallback these values needs to be stored especially when different control instances are having the same host control template. 
Changed the unit tests to match this.